fieldpath/set.go

```diff
@@ -136,6 +138,126 @@ func (s *Set) EnsureNamedFieldsAreMembers(sc *schema.Schema, tr schema.TypeRef)
 	}
 }
 
+// MustPrefixPattern is the same as PrefixPattern except it panics if parts can't be
+// turned into a SetPattern.
+func MustPrefixPattern(parts ...interface{}) *SetPattern {
+	result, err := PrefixPattern(parts...)
+	if err != nil {
+		panic(err)
+	}
+	return result
+}
+
+// PrefixPattern creates a SetPattern that matches all field paths prefixed by the given list of pattern path parts.
+// The pattern parts may any of:
+//
+//   - PathPattern - for wildcards, `MatchAnyPathElement()` can be used as well.
+//   - PathElement - for any path element
+//   - value.FieldList - for associative list keys
+//   - value.Value - for scalar list elements
+//   - string - For field names
+//   - int - for array indices
+func PrefixPattern(parts ...interface{}) (*SetPattern, error) {
+	current := MatchAnySet() // match all field patch suffixes
+	for i := len(parts) - 1; i >= 0; i-- {
+		part := parts[i]
+		var pattern PathPattern
+		switch t := part.(type) {
+		case PathPattern:
+			pattern = t
+		case PathElement:
+			pattern = PathPattern{PathElement: t}
+		case *value.FieldList:
```

Comment:
Is there an example where this is used?  I am trying to understand how it works.

Reply:
I've added some tests.  It's used to define a listMap key field path part (since this system supports dereferecing by list map keys).  Example usage:

```go
NewIncludePatternFilter(MustPrefixPattern("spec", "listMap", &value.FieldList{
  {Name: "key1", Value: value.NewValueInterface("value1")},
  {Name: "key2", Value: value.NewValueInterface("value2")},
})),
```

Reply:
Nice added.